Python/Product/Analysis/LanguageServer/ServerBase.cs

```diff
@@ -20,57 +20,17 @@
 
 namespace Microsoft.PythonTools.Analysis.LanguageServer {
     public abstract class ServerBase : IServer {
-        private RequestLock _lock;
-
-        private sealed class RequestLock : IDisposable {
-            public readonly ServerBase Owner;
-            public readonly CancellationTokenSource CancellationTokenSource;
-
-            public CancellationToken Token => CancellationTokenSource.Token;
-            public void Cancel() => CancellationTokenSource.Cancel();
-
-            public RequestLock(ServerBase owner, int millisecondsTimeout) {
-                CancellationTokenSource = millisecondsTimeout > 0 ?
-                    new CancellationTokenSource(millisecondsTimeout) :
-                    new CancellationTokenSource();
-
-                Owner = owner;
-                if (Interlocked.CompareExchange(ref Owner._lock, this, null) != null) {
-                    throw new InvalidOperationException("currently processing another request");
-                }
-            }
-
-            public void Dispose() {
-                CancellationTokenSource.Dispose();
-                Interlocked.CompareExchange(ref Owner._lock, null, this);
-            }
-        }
-
-        /// <summary>
-        /// Should be used in a using() statement around any requests that support
-        /// cancellation or timeout.
-        /// </summary>
-        public IDisposable AllowRequestCancellation(int millisecondsTimeout = -1) => new RequestLock(this, millisecondsTimeout);
-
-        /// <summary>
-        /// Get this token at the start of request processing and abort when it
-        /// is marked as cancelled.
-        /// </summary>
-        protected CancellationToken CancellationToken => Volatile.Read(ref _lock)?.Token ?? CancellationToken.None;
-
         #region Client Requests
 
-        public abstract Task<InitializeResult> Initialize(InitializeParams @params);
+        public abstract Task<InitializeResult> Initialize(InitializeParams @params, CancellationToken cancellationToken = default(CancellationToken));
```

Comment:
These are also not backwards compatible changes.

Reply:
Maybe I'll just add internal overloads to the `Server` and won't make any changes to public API?